### PR TITLE
JBIDE-27667: Add browser based login

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/LoginWizardPage.java [hunk 1]
```diff
@@ -119,4 +152,52 @@ protected IStatus validate() {
...
+    if (clipboardText == null) {
+      MessageDialog.openError(getWizard().getContainer().getShell(), "Error when parsing login command",
+          "Cannot paste clipboard into the login dialog. Only text is accepted.");
+    } else if (OCCommandUtils.isValidCommand(clipboardText)) {
+      txtURL.setText(OCCommandUtils.getServer(clipboardText));
+      if (IAuthorizationContext.AUTHSCHEME_BASIC.equals(OCCommandUtils.getAuthMethod(clipboardText))) {
+        txtUsername.setText(OCCommandUtils.getUsername(clipboardText));
+        txtPassword.setText(OCCommandUtils.getPassword(clipboardText));
+      } else if (IAuthorizationContext.AUTHSCHEME_OAUTH.equals(OCCommandUtils.getAuthMethod(clipboardText))) {
+        txtToken.setText(OCCommandUtils.getServer(clipboardText));
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/LoginWizardPage.java:168]: please fix issue when pasting login command: 
```suggestion
        txtToken.setText(OCCommandUtils.getToken(clipboardText));
```

**Review by sbouchet — CHANGES_REQUESTED** [review]

**Review by sbouchet — APPROVED** [review]
CDK Unit Test should be fixed, maybe on a separate JIRA. otherwise, LGTM.

**Comment by jeffmaury** [comment]
I don't get why the CDK test are failing where master builds are ok ?

**Comment by jeffmaury** [comment]
testPR

**Comment by jeffmaury** [comment]
testPR

**Pull request merged**